astroplayer-core/native/AstroPlayer/AstroPlayer.swift

```diff
@@ -0,0 +1,159 @@
+import Combine
+import Foundation
+import AVFoundation
+
+@objc public protocol AstroListener { }
+@objc public protocol AstroMediaItem  { }
+@objc public protocol EqualizerValues { }
+
+@objc public enum RepeatMode: Int {
+    case off
+    case all
+    case one
+}
+
+@objc public class iOSAstroPlayer : NSObject {
+    let player = AVQueuePlayer(items: [])
+
+    @objc public var mediaItemCount: Int32 {
+        Int32(player.items().count)
+    }
+
+    @objc public var contentDuration: Int64 = 0
+    @objc public var currentEqualizerValues: EqualizerValues? = nil
+    @objc public var currentMediaItem: AstroMediaItem? = nil
+    @objc public var currentMediaItemIndex: Int32 = 0
+    @objc public var currentPosition: Int64 = 0
+
+    @objc public var isEqualizerEnabled: Bool = false
+    @objc public var isMuted: Bool = false
+    @objc public var isPaused: Bool = false
+    @objc public var isPlaying: Bool = false
+    @objc public var isSmartEqualizerEnabled: Bool = false
```

Comment:
The Swift property:
```swift
@objc public var isSmartEqualizerEnabled: Bool = false
```
gets generated in the KNM as:
```kotlin
@kotlin.commonizer.ObjCCallable
public open expect fun isSmartEqualizerEnabled(): kotlin.Boolean { /* compiled code */ }

@kotlin.commonizer.ObjCCallable
public open expect fun setIsSmartEqualizerEnabled(isSmartEqualizerEnabled: kotlin.Boolean): kotlin.Unit { /* compiled code */ }
```

This discrepancy occurs because the Kotlin/Native cinterop tool generates getter and setter functions for Swift properties instead of direct property access.